core/src/main/java/io/substrait/relation/Extension.java

```diff
@@ -0,0 +1,47 @@
+package io.substrait.relation;
+
+import io.substrait.type.NamedStruct;
+import io.substrait.type.Type;
+import java.util.List;
+
+/** Contains tag interfaces for handling {@link com.google.protobuf.Any} types within Substrait. */
+public class Extension {
+
+  private interface ToProto {
+    com.google.protobuf.Any toProto();
+  }
+
+  public interface Optimization extends ToProto {}
```

Comment:
What's the distinction between Optimization and Enhancement?

Reply:
Explained here: https://github.com/substrait-io/substrait/blob/45b4e483ff1fca3c3e4d0f71e6e55436c6d7638a/proto/substrait/extensions/extensions.proto#L72-L81

Reply:
I've added docs to both `Optimization` and `Enhancement` w/ that information.